tuf/client/updater.py

```diff
@@ -643,6 +643,13 @@ def refresh(self, unsafely_update_root_if_necessary=True):
         logger.info('An expired Root metadata was loaded and must be updated.')
         raise
 
+    # If an exception is raised during the metadata update attempts, we will
+    # attempt to update root metadata once by recursing with a special argument
+    # to avoid further recursion. We use this bool and pull the recursion out
+    # of the except block so as to avoid unprintable nested NoWorkingMirrorError
```

Comment:
I'm curious as to why we are getting an unprintable NoWorkingMirrorError exception.  Is this a bug?  I'm not sure if the `retry_once` boolean is a workaround for the unprintable exceptions.